From: Takashi Iwai <tiwai@suse.de>
Subject: [ALSA] snd_opl3sa2: add missing pnp_unregister_driver() calls
Patch-mainline: 
References: 116586

Add missing pnp_unregister_driver() calls
This fixes Oops at unloading snd-opl3sa2 module.

Signed-off-by: Daniel Ritz <daniel.ritz@gmx.ch>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---


diff -u linux/sound/isa/opl3sa2.c:1.49 linux/sound/isa/opl3sa2.c:1.50
--- linux/sound/isa/opl3sa2.c:1.49	Fri Sep  9 06:21:17 2005
+++ linux/sound/isa/opl3sa2.c	Mon Sep 12 03:47:04 2005
@@ -914,6 +914,7 @@
 #endif
 #ifdef CONFIG_PNP
 		pnp_unregister_card_driver(&opl3sa2_pnpc_driver);
+		pnp_unregister_driver(&opl3sa2_pnp_driver);
 #endif
 		return -ENODEV;
 	}
@@ -927,6 +928,7 @@
 #ifdef CONFIG_PNP
 	/* PnP cards first */
 	pnp_unregister_card_driver(&opl3sa2_pnpc_driver);
+	pnp_unregister_driver(&opl3sa2_pnp_driver);
 #endif
 	for (idx = 0; idx < SNDRV_CARDS; idx++)
 		snd_card_free(snd_opl3sa2_legacy[idx]);
